bundle/edu.gemini.lchquery.servlet/src/main/scala/edu/gemini/lchquery/servlet/LchQueryFunctor.scala

```diff
@@ -0,0 +1,230 @@
+package edu.gemini.lchquery.servlet
+
+import java.io.Serializable
+import java.security.Principal
+import java.text.SimpleDateFormat
+import java.util
+import java.util.{Date, TimeZone}
+import java.util.logging.{Level, Logger}
+
+import edu.gemini.odb.browser._
+import edu.gemini.pot.sp.{ISPNode, ISPObservation, ISPProgram}
+import edu.gemini.pot.spdb.{DBAbstractQueryFunctor, IDBDatabaseService, IDBFunctor, IDBParallelFunctor}
+import edu.gemini.skycalc.{DDMMSS, HHMMSS}
+import edu.gemini.spModel.gemini.obscomp.SPSiteQuality
+import edu.gemini.spModel.obs.{ObsClassService, ObservationStatus}
+import edu.gemini.spModel.rich.shared.immutable._
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.spModel.target.env.TargetEnvironment
+import edu.gemini.spModel.too.Too
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+
+class LchQueryFunctor(queryType: LchQueryFunctor.QueryType,
+                      programParams: List[(LchQueryParam[ISPProgram], String)],
+                      observationParams: List[(LchQueryParam[ISPObservation], String)])
+  extends DBAbstractQueryFunctor with IDBParallelFunctor {
+
+  // The current QueryResult. Must be mutable.
+  var queryResult: QueryResult = new QueryResult() {
```

Comment:
I think it's a consequence of being an `IDBParallelFunctor`, which requires the implementation of the `mergeResults` method, which has to recreate the object.

Honestly, I'm not even convinced that we need this, because based on what I saw in the old code, I am thinking that this method would fail with a `NullPointerException` so it was probably never called: when you create new instance of `QueryResult`, if you don't also set it to have an instance of `ProgramsNode`, this is set to `null` by default.

In the old implementation, a new `QueryResult` was created but no new `ProgramsNode` was set to it, so the call `queryResult.getProgramsNode.getPrograms` would have thrown a `NullPointerException` and failed.

Thus, maybe we can remove this and the implementation of `IDBParallelFunctor` and instead change `queryResult` to an immutable instance of `QueryResult`, which is obviously mutable.

@swalker2m - do you have any thoughts on this?